src/Language/Java/Inline.hs

```diff
@@ -349,8 +352,7 @@ blockQQ input = case Java.parser Java.block input of
 #else
             TH.VarI _ ty _ _ -> do
 #endif
-              -- First (recursively) unfold type synonyms, if any.
-              ty' <- unfoldHeadTySyn ty
+              ty' <- unfoldTypeTySyn ty
               case ty' of
                 TH.AppT (TH.ConT nJ) thty
                   | nJ == ''J -> do
```

Comment:
`unliftJType` is called below this line on a parameter where type synonyms are already unfolded. Probably `unliftJType` shouldn't call `unfoldTypeTySyn`.